flutter_local_notifications/ios/Classes/FlutterLocalNotificationsPlugin.m

```diff
@@ -234,6 +252,78 @@ - (void)pendingNotificationRequests:(FlutterResult _Nonnull)result {
   }
 }
 
+/// Extracts notification categories from [arguments] and configures them as
+/// appropriate.
+///
+/// This code will simply return the `completionHandler` if not running on a
+/// compatible iOS version or when no categories were specified in [arguments].
+- (void)configureNotificationCategories:(NSDictionary *_Nonnull)arguments
+                  withCompletionHandler:(void (^)(void))completionHandler {
+  if (@available(iOS 10.0, *)) {
+    if ([self containsKey:@"notificationCategories" forDictionary:arguments]) {
+      NSMutableSet<UNNotificationCategory *> *newCategories =
+          [NSMutableSet set];
+
+      NSArray *categories = arguments[@"notificationCategories"];
+
+      for (NSDictionary *category in categories) {
+        NSMutableArray<UNNotificationAction *> *newActions =
+            [NSMutableArray array];
+
+        NSArray *actions = category[@"actions"];
+        for (NSDictionary *action in actions) {
+          NSString *type = action[@"type"];
+          NSString *identifier = action[@"identifier"];
+          NSString *title = action[@"title"];
+          UNNotificationActionOptions options =
+              [Converters parseNotificationActionOptions:action[@"options"]];
+
+          if ([type isEqualToString:@"plain"]) {
+            [newActions
+                addObject:[UNNotificationAction actionWithIdentifier:identifier
+                                                               title:title
+                                                             options:options]];
+          } else if ([type isEqualToString:@"text"]) {
+            NSString *buttonTitle = action[@"buttonTitle"];
+            NSString *placeholder = action[@"placeholder"];
+            [newActions addObject:[UNTextInputNotificationAction
+                                      actionWithIdentifier:identifier
+                                                     title:title
+                                                   options:options
+                                      textInputButtonTitle:buttonTitle
+                                      textInputPlaceholder:placeholder]];
+          }
+        }
+
+        UNNotificationCategory *newCategory = [UNNotificationCategory
+            categoryWithIdentifier:category[@"identifier"]
+                           actions:newActions
+                 intentIdentifiers:@[]
+                           options:[Converters parseNotificationCategoryOptions:
+                                                   category[@"options"]]];
+
+        [newCategories addObject:newCategory];
+      }
+
+      if (newCategories.count > 0) {
+        UNUserNotificationCenter *center =
+            [UNUserNotificationCenter currentNotificationCenter];
+        [center getNotificationCategoriesWithCompletionHandler:^(
```

Comment:
@ened I figured out what was the cause behind why notification actions stopped appearing for me on iOS and narrowed it to here. removing the logic to query the existing categories and then appending to that list allowed me to see the notification actions again. was there a reason for querying this? from my own tests and what i've seen in docs, should be able to just call `[center setNotificationCategories:newCategories];`